### PR TITLE
1271 ability drafted

### LINKED ISSUE #1271: Ability draft skills on match page
In ability draft matches ([this one, for example](https://www.opendota.com/matches/3509547800)), it'd be nice to have a visual display of which skills each hero drafted (instead of trying to parse each skill out of the ability build chart, as I do now) somewhere on the match's page.

### PR DESCRIPTION
**Pull request by infinius12345**
fixes #1271

### CODE REVIEW

File: src/actions/transformMatch.js [hunk 1]
```diff
@@ -211,6 +211,16 @@ function transformMatch(m) {
     }
     newPlayer.buybacks = (player.buyback_log || []).length;
     newPlayer.total_gold = (player.gold_per_min * m.duration) / 60;
+    if (m.game_mode === 18) {
+      const arr = [];
+      arr[0] = 1;
```
Comment (howardchung) [inline src/actions/transformMatch.js]: what is 1?

File: src/actions/transformMatch.js [hunk 2]
```diff
@@ -211,6 +211,16 @@ function transformMatch(m) {
...
+      player.ability_upgrades_arr.forEach((ability) => {
```
Comment (howardchung) [inline src/actions/transformMatch.js]: check to make sure player.ability_upgrades_arr exists

File: src/components/Match/matchColumns.jsx [hunk 3]
```diff
@@ -294,6 +294,25 @@ export const abilityColumns = () => {
   return cols;
 };
 
+export const abilityDraftColumns = () => {
+  const cols = Array.from(new Array(6), (_, index) => ({
+    displayName: `${index}`,
+    tooltip: 'Ability Drafted',
```
Comment (howardchung) [inline src/components/Match/matchColumns.jsx]: Move this to string file

File: src/lang/en-US.json [hunk 4]
```diff
@@ -283,6 +283,7 @@
   "heading_graph_xp": "Experience",
   "heading_graph_lh": "Last Hits",
   "heading_overview": "Overview",
+  "heading_ability_draft": "Abilites Drafted",
```
Comment (howardchung) [inline src/lang/en-US.json]: "Abilities"

**Comment by howardchung** [comment]
Can you give an example match ID so we can see how this looks?

**Comment by infinius12345** [comment]
/matches/3509547800 ->example

File: src/actions/transformMatch.js (hunk 1 above)
Comment (infinius12345) [reply src/actions/transformMatch.js]: The table doesnt show the index[0] of the array, so i just pushed a dummy value

**Comment by GravityMsc** [comment]
@infinius12345  You should run `npm run lint ` to correct your code by the rules firstly.

**Comment by infinius12345** [comment]
image

this is what i get .. all errors are the same and most are not my files

**Comment by Argn0** [comment]
you need to fix these

Overview.jsx
  8:8   error  A space is required after '{'   object-curly-spacing
  8:61  error  A space is required before '}'  object-curly-spacing

**Comment by infinius12345** [comment]
Tyvm - idk why it wasnt showing up on my npm run lint

File: src/actions/transformMatch.js [hunk 5]
```diff
@@ -211,6 +211,16 @@ function transformMatch(m) {
     }
     newPlayer.buybacks = (player.buyback_log || []).length;
     newPlayer.total_gold = (player.gold_per_min * m.duration) / 60;
+    if (m.game_mode === 18 && Object.prototype.hasOwnProperty.call(player, 'ability_upgrades_arr')) {
+      const arr = [];
+      arr[0] = 1;
```
Comment (howardchung) [inline src/actions/transformMatch.js]: instead of creating a dummy element here, perhaps just refer to the `i+1`th element in the column definition?

Comment (infinius12345) [reply src/actions/transformMatch.js]: I did it, just referring the the -1 index of incoming array.. let me know if that is a bad soln.

**Pull request merged**